pages/account.js

```diff
@@ -0,0 +1,255 @@
+import React, {Component} from 'react';
+import NavbarBootstrap from '../components/NavbarBootstrap';
+import Container from 'react-bootstrap/Container';
+import Card from 'react-bootstrap/Card';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Nav from 'react-bootstrap/Nav';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import CarRow from '../components/CarRow';
+import axios from 'axios';
+
+const host = 'http://localhost';
+const port = 3001;
+
+class Account extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isAuthenticated: false,
+            name: '',
+            email: '',
+            newName: '',
+            newEmail: '',
+            succeededProfileUpdate: false,
+            failedProfileUpdate: false,
+            oldPasswordInput: '',
+            newPasswordInput: '',
+            succeededPasswordUpdate: false,
+            failedPasswordUpdate: false,
+        };
+    }
+
+    componentDidMount = () => {
+        const authToken = window.localStorage.getItem('auth_token');
+
+        if (!authToken) {
+            this.setState({ isAuthenticated: false });
+        } else {
+            axios.post(`${host}:${port}/authstatus`, { headers: { 'Authorization' : `Bearer ${authToken}` } }).then(res => {
+                if (res.status === 200) {
+                    const { name, email } = res.data.authData;
+                    this.setState({
+                        isAuthenticated: true,
+                        name,
+                        email,
+                        newName: name,
+                        newEmail: email,
+                    });
+                }
+            }).catch(err => {});
+        }
+    }
+
+    updateProfile = () => {
+        const { email, newName, newEmail } = this.state;
+
+        axios.post(`${host}:${port}/updateProfile`, {
+            oldEmail: email,
+            newName,
+            newEmail,
+        })
+        .then(response => {
+            const { newName, newEmail, token } = response.data;
+            window.localStorage.setItem('auth_token', token);
+            this.setState({ name: newName, email: newEmail, newName, newEmail, succeededProfileUpdate: true });
+        })
+        .catch(error => {
+            console.log(error);
+            this.setState({ failedProfileUpdate: true });
+        });
+    }
+
+    updatePassword = () => {
+        const { email, oldPassword, newPassword } = this.state;
```

Comment:
we need to add a check here to make sure that the old and new password fields are not empty
it is a waste of resources to make a network request just to know that the user clicked updatepassword without even entering anything